sale_order_import/wizard/sale_order_import.py

```diff
@@ -0,0 +1,667 @@
+    )
+    partner_id = fields.Many2one("res.partner", string="Customer")
+    import_type = fields.Selection(
+        [("xml", "XML"), ("pdf", "PDF")],
+        required=True,
+        default=None,
+        help="Select a type which you want to import",
+    )
+    order_file = fields.Binary(
+        string="Request for Quotation or Order",
+        required=True,
+        help="Upload a Request for Quotation or an Order file. Supported "
+        "formats: XML and PDF (PDF with an embeded XML file).",
+    )
+    order_filename = fields.Char(string="Filename")
+    doc_type = fields.Selection(
+        [("rfq", "Request For Quotation"), ("order", "Sale Order")],
+        string="Document Type",
+        readonly=True,
+    )
+    price_source = fields.Selection(
+        [("pricelist", "Pricelist"), ("order", "Customer Order")],
+        string="Apply Prices From",
+    )
+    # for state = update
+    commercial_partner_id = fields.Many2one(
+        "res.partner", string="Commercial Entity", readonly=True
+    )
+    partner_shipping_id = fields.Many2one(
+        "res.partner", string="Shipping Address", readonly=True
+    )
+    sale_id = fields.Many2one("sale.order", string="Quotation to Update")
+
+    @api.onchange("order_file")
+    def order_file_change(self):
+        if not self.order_filename or not self.order_file:
+            self.doc_type = False
+            return
+
+        doc_type = self._parse_file(
+            self.order_filename, b64decode(self.order_file), detect_doc_type=True
+        )
+        if doc_type is None:
+            return {"warning": self._unsupported_file_msg(self.order_filename)}
+        self.doc_type = doc_type
+
+    def _parse_file(self, filename, filecontent, detect_doc_type=False):
+        assert filename, "Missing filename"
+        assert filecontent, "Missing file content"
+        filetype = mimetypes.guess_type(filename)
+        logger.debug("Order file mimetype: %s", filetype)
+        mimetype = filetype[0]
+        supported_types = {
+            "xml": ("application/xml", "text/xml"),
+            "pdf": ("application/pdf"),
+        }
+        # Check if the selected import type is supported
+        if self.import_type not in supported_types:
+            raise UserError(_("Please select a valid import type before importing!"))
+
+        # Check if the detected MIME type is supported for the selected import type
+        if mimetype not in supported_types[self.import_type]:
+            raise UserError(
+                _(
+                    "This file '{filename}' is not recognized as a {type} file. "
+                    "Please check the file and its extension.",
+                    filename=filename,
+                    type=self.import_type.upper(),
+                )
+            )
+        # Call the appropriate parsing method based on the selected import type
+        if self.import_type == "xml":
```

Comment:
@QuocDuong1306 this can be extracted to a method, people can inherit then for deal with other import types